### ISSUE TITLE
Window cursor with negative colors

### ISSUE THREAD

**Issue by blackhamm3rjack** [SFML/SFML#1464]
## Subject of the issue

Whenever I try to load a window icon over the Windows platform, using a 32-bits PNG image, the resulting colors are inverted. I determined that the color format of the image I use is RGBA meanwhile the correct format for a window cursor is BGRA. I then inverted the colors through a loop that changes every pixel in the image and the problem was fixed.

In the documentation there's written that the required format is RGBA, but I find it wrong. I don't know if it's an issue related to how the Windows API manages cursors or if I'm totally wrong.

## Your environment

* Windows 10 x64 Version 10.0.17134 Build 17134
* SFML from git master
* Microsoft (R) C/C++ Optimizing Compiler Version 19.14.26433 for x64
* No special compiler flags

## Steps to reproduce

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(1280, 720), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    // Load the icon
    sf::Image image;
    sf::Cursor cursor;
    if (!image.loadFromFile("icon.png"))
        return -1;
    if (!cursor.loadFromPixels(image.getPixelsPtr(), image.getSize(), { 0, 0 }))
        return -1;
    window.setMouseCursor(cursor);

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```

## Expected behavior

The window cursor should be displayed with the correct image colors.

## Actual behavior

The cursor image appears as if the colors were negated.

## Fix

```cpp
// Swap the color format from RGBA to BGRA
for (unsigned x = 0; x < image.getSize().x; ++x)
{
	for (unsigned y = 0; y < image.getSize().y; ++y)
	{
		sf::Color pixel = image.getPixel(x, y);
		image.setPixel(x, y, { pixel.b, pixel.g, pixel.r, pixel.a });
	}
}
```

**Comment by blackhamm3rjack** [comment]
Example cursor: ![arrow](https://user-images.githubusercontent.com/6911442/43892352-f92797c8-9bcb-11e8-929d-68608c8c5a6c.png)

**Comment by MarioLiebisch** [comment]
Branch `bugfix/windows-cursor-colors` should fix this, give it a spin, if you can.

**Comment by eXpl0it3r** [comment]
While the proposed fix might fix it, I still wonder where the information can be found that BGRA is expected?

**Comment by eXpl0it3r** [comment]
Fixed with #1465
